cjworkbench/i18n/trans.py

```diff
@@ -209,6 +212,34 @@ def for_application(self) -> MessageLocalizer:
         """Return a `MessageLocalizer` for the application messages"""
         return self._app_localizer
 
+    @lru_cache(1)
+    def for_cjwmodule(self) -> MessageLocalizer:
+        """Return a `MessageLocalizer` for the messages of `cjwmodule`"""
+        catalogs = {}
```

Comment:
I would expect `for_cjwmodule()` and `for_application()` to share design patterns, since their behavior is so very similar. For instance, if `for_application` uses `self._app_localizer()`, then `for_cjwmodule()` should use `self._cjwmodule_localizer()`.

Please nix `cjworkbench/i18n/catalogs/__init__.py` and move its functions into here. Nix `BadCatalogsError` -- everywhere it occurs, we actually want a crash. And nix `UnsupportedLocaleError` because it wraps an `Exception` and I'd rather we have the underlying exception.

To clarify, here is the behavior we should be having here:

* If `application` is missing a pofile even though we mark it a supported locale, _crash_. (Simply ... don't catch anything. I'd expect a `FileNotFoundError`; if it's something else, maybe add a comment to that effect.)
* If `cjwmodule` is missing a pofile, log a warning and supply an empty catalog, so translations will use the default locale. (This will help us support new languages out-of-order -- i.e., translate `cjworkbench` before translating `cjwmodule`.)
* If `application` _or_ `cjwmodule` contains an invalid pofile, _crash_. (Don't catch anything.)

Reply:
Some of the stuff in `cjworkbench/i18n/catalogs/__init__.py` is also used in scripts under `cjworkbench/i18n/catalogs`